rewrite-java/src/main/java/org/openrewrite/java/format/SpacesVisitor.java

```diff
@@ -1206,7 +1206,7 @@ public J.TypeParameter visitTypeParameter(J.TypeParameter typeParam, P p) {
     @Override
     public J postVisit(J tree, P p) {
         if (stopAfter != null && stopAfter.isScope(tree)) {
-            getCursor().putMessageOnFirstEnclosing(JavaSourceFile.class, "stop", true);
+            getCursor().getRoot().putMessage("stop", true);
```

Comment:
Doesn't the visitor need to make sure to at some point clear the message again before returning control to the caller? In the `visit()` method the message is only read, but not cleared.

Reply:
The cursor instance itself will be garbage collected after the visit returns. Cursor is a telescoping data type, so it unwinds as it moves back up the stack and the pieces that are getting popped off are dereferenced (along with any messages on them).